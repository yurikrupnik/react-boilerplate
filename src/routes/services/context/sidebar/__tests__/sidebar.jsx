import React from 'react';
import { Consumer, Provider } from '../index';

const {
    describe,
    test,
    expect,
    shallow,
    render
} = global;

describe('Device context', () => {
    const Wrapper = () => ((
        <div>hello</div>
    ));
    test('render consumer', () => {
        const wrapper = shallow( // notice render here
            <Consumer render={() => (
                <Wrapper />
            )}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });
    test('render provider', () => {
        const wrapper = shallow((
            <Provider>
                <Wrapper />
            </Provider>
        ));
        expect(wrapper).toMatchSnapshot();
    });
    // test('defualt props', () => {
    //     const wrapper = shallow((
    //         <Provider userAgent={navigator.userAgent}>
    //             <Wrapper />
    //         </Provider>
    //     ));
    //     expect(wrapper).toMatchSnapshot();
    // });

    test('dynamic consumer', () => {
        const wrapper = shallow((
            <Provider>
                <Consumer render={(props) => {
                    return (
                        <Wrapper />
                    );
                }}
                />
            </Provider>
        ));
        expect(wrapper).toMatchSnapshot();
    });
    test('default consumer', () => {
        const wrapper = render((
            <Provider>
                <Consumer links={[
                    {
                        key: '1',
                        to: '1',
                        label: '1'
                    }
                ]}
                />
            </Provider>
        ));
        expect(wrapper).toMatchSnapshot();
    });
    test('methods', () => {
        const wrapper = render((
            <Provider>
                <Consumer render={(props) => {
                    props.toggle();
                    return (
                        <Wrapper />
                    );
                }}
                />
            </Provider>
        ));
        expect(wrapper).toMatchSnapshot();
    });
});
