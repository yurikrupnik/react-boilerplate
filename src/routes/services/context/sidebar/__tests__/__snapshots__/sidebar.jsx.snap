// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Device context default consumer 1`] = `null`;

exports[`Device context dynamic consumer 1`] = `
<ContextProvider
  value={
    Object {
      "open": false,
      "toggle": [Function],
    }
  }
>
  <SidebarConsumer
    render={[Function]}
  />
</ContextProvider>
`;

exports[`Device context methods 1`] = `
<div>
  hello
</div>
`;

exports[`Device context render consumer 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Device context render provider 1`] = `
<ContextProvider
  value={
    Object {
      "open": false,
      "toggle": [Function],
    }
  }
>
  <Wrapper />
</ContextProvider>
`;
